diff --git a/node_modules/react-native-screens/cpp/RNSScreenRemovalListener.cpp b/node_modules/react-native-screens/cpp/RNSScreenRemovalListener.cpp
index e634a46..d96e808 100644
--- a/node_modules/react-native-screens/cpp/RNSScreenRemovalListener.cpp
+++ b/node_modules/react-native-screens/cpp/RNSScreenRemovalListener.cpp
@@ -10,8 +10,7 @@ std::optional<MountingTransaction> RNSScreenRemovalListener::pullTransaction(
   for (const ShadowViewMutation &mutation : mutations) {
     if (mutation.type == ShadowViewMutation::Type::Remove &&
         mutation.oldChildShadowView.componentName != nullptr &&
-        strcmp(mutation.parentShadowView.componentName, "RNSScreenStack") ==
-            0) {
+        strcmp(mutation.oldChildShadowView.componentName, "RNSScreen") == 0) {
       listenerFunction_(mutation.oldChildShadowView.tag);
     }
   }
diff --git a/node_modules/react-native-screens/ios/RNSScreenStack.mm b/node_modules/react-native-screens/ios/RNSScreenStack.mm
index fe9dc57..d69d546 100644
--- a/node_modules/react-native-screens/ios/RNSScreenStack.mm
+++ b/node_modules/react-native-screens/ios/RNSScreenStack.mm
@@ -1221,7 +1221,7 @@ - (void)mountingTransactionDidMount:(const facebook::react::MountingTransaction
                withSurfaceTelemetry:(const facebook::react::SurfaceTelemetry &)surfaceTelemetry
 {
   for (const auto &mutation : transaction.getMutations()) {
-    if (mutation.parentShadowView.tag == self.tag &&
+    if (mutation.parentTag == self.tag &&
         (mutation.type == react::ShadowViewMutation::Type::Insert ||
          mutation.type == react::ShadowViewMutation::Type::Remove)) {
       // we need to wait until children have their layout set. At this point they don't have the layout
